LPC Projects/pwmdemodulator/src/pwmdemodulator.c

```diff
@@ -0,0 +1,300 @@
+/*
+===============================================================================
+ Name        : pwmdemodulator.c
+ Author      : $(author)
+ Version     :
+ Copyright   : $(copyright)
+ Description : main definition
+===============================================================================
+*/
+
+#if defined (__USE_LPCOPEN)
+#if defined(NO_BOARD_LIB)
+#include "chip.h"
+#else
+#include "board.h"
+#endif
+#endif
+
+#include <cr_section_macros.h>
+
+#define TEST_CCAN_BAUD_RATE 1000000
+CCAN_MSG_OBJ_T msg_obj;
+
+uint16_t k_min_pulse = 500;
+uint16_t k_max_pulse = 2500;
+uint16_t k_offset_rc_in = 1500;
+uint16_t k_scale_rc_in = 350;
+uint16_t k_offset_stored_angle = 0;
+uint16_t k_scale_stored_angle = 1500;
+
+uint16_t RiseTime = 0;
+uint16_t PulseWidth_US = 0;
+
+uint16_t BrakeRiseTime = 0;
+bool brake = true;
+
+uint32_t AutonRiseTime = 0;
+bool auton = false;
+
+void CAN_IRQHandler(void) {
+	LPC_CCAN_API->isr();
+}
+
+void CAN_tx(uint8_t msg_obj_num) {
+	Board_UARTPutSTR("Sent\n");
+}
+
+void CAN_rx(uint8_t msg_obj_num) {
+	/* Determine which CAN message has been received */
+	msg_obj.msgobj = msg_obj_num;
+	/* Now load up the msg_obj structure with the CAN message */
+	LPC_CCAN_API->can_receive(&msg_obj);
+	if (msg_obj_num == 1) {
```

Comment:
Do we want to echo every CAN message we receive?

Reply:
oops that was my bad, forgot to remove this